sycl/include/CL/sycl/backend/level_zero.hpp

```diff
@@ -9,7 +9,8 @@
 #pragma once
 
 #include <CL/sycl.hpp>
-#include <level_zero/ze_api.h>
+// This header should be included by users.
+//#include <level_zero/ze_api.h>
```

Comment:
Is there any "standard" macro defined in `level_zero/ze_api.h`, which we can check?
It would be better to have something like:
```c++
#if !defined(LEVEL_ZERO_INCLUDED)
  #error "level_zero/ze_api.h must be included to use Level Zero interoperability with SYCL"
#endif
```
instead of these two lines.

Reply:
There is probably soemthing we could use, e.g. _ZE_API_H, but not sure how "standard" this is. Probably not (+@jandres742)
Also, since all the uses of L0 types are in templates, it is probably fine to have L0 included after this header

Reply:
I recalled that we tried the same approach with OpenCL headers in the past and it caused some issues, so we include OpenCL headers from SYCL headers.
What is the motivation for this change? Will it be clear for user that includes of Level Zero headers are missing and how to fix this problem?

Reply:
The ze_api.h has this

#ifndef _ZE_API_H

but not quite sure what we are trying to solve here. Why we cannot include the ze_api.h in the interop header? We know the interop header is gonna need it anyway, no?

Reply:
If I understand it correctly, this patch is supposed to address the compile time issue with missing headers on the systems w/o Intel GPU (e.g. cpu-only system or a system with NVIDIA GPU). In these cases DPC++ is not supposed to use Level Zero plug-in and there might be no headers on the system.

I'm not sure that asking users to provide headers in the best option here, but it's one way to address this issue.

Reply:
@bader You are right.